lua/lspconfig/bashls.lua

```diff
@@ -7,7 +8,17 @@ configs[server_name] = {
   default_config = {
     cmd = {"bash-language-server", "start"};
     filetypes = {"sh"};
-    root_dir = util.path.dirname;
+    root_dir = function(fname)
+      local dirname = util.path.dirname(fname)
+
+      -- Prevent the bash-language-server from scanning the entire home folder.
+      if dirname == vim.loop.os_homedir() then
+        log.info('Not setting root_dir to prevent scanning everything in $HOME')
```

Comment:
Not sure how to phrase this in a good way... or if we should bother.

Reply:
This is probably fine. You tested the log right? 

Maybe change the info to:
`Detected root directory is $HOME will will hang bashls. Move your file to to a subdirectory and reopen.`

Reply:
And maybe note this in the comment
```
  -- Prevent the bash-language-server from hanging while scanning the entire home folder.
```

Reply:
> This is probably fine. You tested the log right?

Yeah, except the default log level is warn(ing), so most people will likely miss it. I could change it to `log.warn` though.

But maybe the default log level should be changed instead? https://github.com/neovim/neovim/blob/1785ac3e3787e84846d753ceb73a239f5575a691/runtime/lua/vim/lsp/log.lua#L22

Reply:
I would just change it to warn right now, this is kinda a "warning" in that the language server won't start. Has this issue been filed upstream? I tried searching the bashls issue tracker and didn't see anything. There should be an option not to index home directory given some people are going to try to edit bashrc/profile etc.